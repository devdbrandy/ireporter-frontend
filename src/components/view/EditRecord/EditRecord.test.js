import React from 'react';
import { shallow } from 'enzyme';
import { EditRecord } from './index';

/**
 * Wrapper for enzyme shallow component
 *
 * @returns {object} The mocked props and component
 */
const setup = () => {
  const props = {
    createRecord: {},
    userRecords: [],
    updateRecord: jest.fn(),
  };

  const component = shallow(<EditRecord {...props} />);
  return { props, component };
};

describe('<EditRecord />', () => {
  it('renders without crashing given the required props', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
