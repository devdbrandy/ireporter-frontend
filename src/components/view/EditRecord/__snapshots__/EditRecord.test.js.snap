// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditRecord /> renders without crashing given the required props 1`] = `
<Fragment>
  <Layout>
    <div
      className="container"
    >
      <main
        id="main"
      >
        <div
          className="column"
        >
          <div
            className="panel"
          >
            <div
              className="panel--heading"
            >
              <div
                className="row"
              >
                <h3>
                  Add New Record
                </h3>
                <div
                  className="column"
                >
                  <div
                    className="row end"
                  >
                    <button
                      className="btn btn-back"
                      type="button"
                    >
                      <i
                        className="fas fa-reply"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="panel--body"
            >
              <form
                id="create-record-form"
                onSubmit={[Function]}
              >
                <div
                  className="form-control"
                >
                  <div
                    className="col-2"
                  >
                    <label
                      htmlFor="#"
                    >
                      Record Type
                    </label>
                  </div>
                  <div
                    className="col-10"
                  >
                    <label>
                      <input
                        checked={false}
                        name="type"
                        onChange={[Function]}
                        type="radio"
                        value="red-flags"
                      />
                      Red Flag
                    </label>
                    <label>
                      <input
                        checked={false}
                        name="type"
                        onChange={[Function]}
                        type="radio"
                        value="interventions"
                      />
                      Intervention
                    </label>
                  </div>
                </div>
                <div
                  className="form-control"
                >
                  <div
                    className="col-2"
                  >
                    <label
                      htmlFor="title"
                    >
                      Short Title
                    </label>
                  </div>
                  <div
                    className="col-10"
                  >
                    <input
                      id="title"
                      name="title"
                      onChange={[Function]}
                      placeholder="Enter a short title"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="form-control"
                >
                  <div
                    className="col-2"
                  >
                    <label
                      htmlFor="comment"
                    >
                      Comment
                    </label>
                  </div>
                  <div
                    className="col-10"
                  >
                    <textarea
                      id="comment"
                      name="comment"
                      onChange={[Function]}
                      placeholder="Enter record comment"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="form-control"
                >
                  <div
                    className="col-2"
                  >
                    <label
                      htmlFor="geoautocomplete"
                    >
                      Location
                    </label>
                  </div>
                  <PlacesAutocomplete
                    debounce={200}
                    highlightFirstSuggestion={false}
                    onChange={[Function]}
                    onError={[Function]}
                    onSelect={[Function]}
                    searchOptions={Object {}}
                    shouldFetchSuggestions={true}
                    value=""
                  >
                    <Component />
                  </PlacesAutocomplete>
                </div>
                <div
                  className="form-control"
                >
                  <div
                    className="col-2"
                  />
                  <div
                    className="col-10"
                  >
                    <div
                      className="upload-btn-wrapper"
                    >
                      <label>
                        <button
                          className="cloudinary-button"
                          id="upload-widget"
                          onClick={[Function]}
                          type="button"
                        >
                          <i
                            className="fas fa-paperclip"
                          />
                        </button>
                        <input
                          id="media"
                          name="media"
                          type="hidden"
                          value="[]"
                        />
                         
                        <small>
                          Upload media files
                        </small>
                      </label>
                    </div>
                    <ul
                      className="media-list"
                    />
                  </div>
                </div>
                <div
                  className="form-control"
                >
                  <div
                    className="col-2"
                  >
                    <label
                      htmlFor="status"
                    >
                      Status
                    </label>
                  </div>
                  <div
                    className="col-10"
                  >
                    <label>
                      <input
                        checked={true}
                        name="status"
                        onChange={[Function]}
                        type="radio"
                        value="draft"
                      />
                      Draft
                    </label>
                    <label>
                      <input
                        checked={false}
                        name="status"
                        onChange={[Function]}
                        type="radio"
                        value="published"
                      />
                      Publish
                    </label>
                  </div>
                </div>
                <div
                  className="row end mt-60"
                >
                  <button
                    className="btn btn-primary"
                    type="submit"
                  >
                    Update record
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </main>
    </div>
  </Layout>
</Fragment>
`;
